packages/common/src/utils/conditionalFormatting.ts

```diff
@@ -101,6 +101,12 @@ export const hasMatchingConditionalRules = (
     const parsedValue = typeof value === 'string' ? Number(value) : value;
     const convertedValue = convertFormattedValue(parsedValue, field);
 
+    const currentFieldId = getItemId(field);
+    const targetFieldId = config.target?.fieldId;
+    if (targetFieldId !== undefined && targetFieldId !== currentFieldId) {
```

Comment:
if rule target is undefined or target equals to current cell's field id this function should not stop